### DISCUSSION TITLE
ADJOINT problem and RANS/NAVIER_STOKES solver does not work

### DISCUSSION THREAD

**Discussion by CipollinoDuck** [su2code/SU2#1456] [Q&A]
Hello everyone,

I ask here because I cannot find any solution elsewhere.

I am attempting an **Adjoint Optimization problem** with viscous solver (**RANS/NAVIER_STOKES**) using the **shape_optimization.py** script but I can't complete it because I receive the following message:

`Error in "void CSolver::Read_SU2_Restart_Binary(CGeometry*, const CConfig*, std::__cxx11::string)":`
`-------------------------------------------------------------------------`
`Unable to open SU2 restart file solution_adj_cd.dat`

With EULER solver it works fine, it creates the **solution_adj_*.dat** files everytime it runs the ADJOINT problem so, at the beginning, the process does not require any further file besides the .su2 and the .cfg ones.

Even the _Constrained shape design of a transonic turbulent airfoil tutorial_ does not work. The required files (_initial solution files for the solver and adjoint solution_flow.dat, solution_adj_cd.dat, and solution_adj_cmz.dat_) are given from the beginning and if they are not added to the path in which the tutorial will be run, it cannot be completed.

I also tried the **turb_ONERAM6 Test case** and personal files as well.

Furthermore, I tried to cheat (?) creating a solution_adj_cd.dat and solution_adj_cmz.dat files from the EULER ADJOINT problem because it was the only way I had to obtain that file: the RANS ADJOINT problem performed better than expected reaching a feasible geometry which is not the one obtained in the tutorial.

I do not get why I have this problem, if it is correct or not. Has anyone experienced this before?

Actually, I think I will run this kind of problem with the cheating (?) procedure described before because I need to go ahead with the work and it seems to work relatively fine.

In conclusion, the **ADJOINT** problem with **RANS/NAVIER_STOKES** solver does not work because it needs **solution_adj_*.dat** which are created autonomously with EULER solver.

I am using SU2 V7.2.1 but I also tested them with versions from this to 7.0.0.

The files from the tutorial without solution_*.dat files:
[turb_SA_RAE2822.zip](https://github.com/su2code/SU2/files/7639785/turb_SA_RAE2822.zip)

Personal files:
[naca0012_8_1_0_700.zip](https://github.com/su2code/SU2/files/7640051/naca0012_8_1_0_700.zip)

The command that I use is: python shape_optimization.py -n 8 -g CONTINUOUS_ADJOINT -o SLSQP -f used_file.cfg

UPDATE: I also compiled the code in Ubuntu 20.04 enabling the DISCRETE_ADJOINT functionality but I always receive the same message.

How can I create those solution_adj_*.dat files to start the ADJOINT problem?

Have I missed something written in other tutorials?

Thank you very much for all of the support.

**Comment by pcarruscag** [comment]
I can't unzip either of your files.
I tried the RAE2822 tutorial and it works with both DISCRETE_ADJOINT and CONTINUOUS_ADJOINT.

**Reply by CipollinoDuck** [reply]
Hi Pedro,
Thank you for your reply.

I used WinRAR just now and the unzip worked fine even on my phone. 

Have you tried to run the tutorial without the restart_adjoint files? I ask you this because I have found the problem without using them since they are case sensitive. I found that I am not the only one facing this problem. 

I managed to run the simulations running a Direct solution using RANS/NAVIER-STOKES solvers and a first iteration of the CONTINUOUS/DISCRETE ADJOINT problem using EULER solver for every constraint function used. With this procedure I generated the restart_adjoint files to run the RANS/NS adjoint problem.

Is there any regular procedure to generate those files since neither shape_optimization.py and the manual procedure can?

**Reply by pcarruscag** [reply]
Ok I understand the issue now:
The NS continuous adjoint solver tries to load a restart even with RESTART_SOL = NO.
It will be fixed shortly in #1476
